empyrical/stats.py

```diff
@@ -140,11 +142,14 @@ def cum_returns(returns, starting_value=0):
     if len(returns) < 1:
         return type(returns)([])
 
-    if np.isnan(np.asanyarray(returns)[0]):
-        returns = returns.copy()
-        returns[0] = 0.
+    if isinstance(returns, pd.DataFrame):
+        df_cum = (returns.fillna(0) + 1).cumprod()
 
-    df_cum = np.exp(nancumsum(np.log1p(returns)))
+    else:
+        if np.all(np.isnan(np.asanyarray(returns)[0])):
```

Comment:
Sorry, no. This is for the case where a user passes in either a `pd.Series().pct_change()` or a `pd.DataFrame().pct_change()` of prices. The current `all()` catches both of those cases. `any()`, as I understood you were proposing, would catch a case that we do not anticipate.